### ISSUE TITLE
Mock Page

### ISSUE THREAD

**Issue by OleVik** [getgrav/grav#2682]
The current Page - implementing [PageInterface](https://github.com/getgrav/grav/blob/develop/system/src/Grav/Common/Page/Interfaces/PageInterface.php) - relies heavily on the [PageLegacyInterface](https://github.com/getgrav/grav/blob/develop/system/src/Grav/Common/Page/Interfaces/PageLegacyInterface.php#L19), where the initiation of the Page is almost entirely dependent on the existence of a physical file because of the `init(\SplFileInfo $file, $extension = null)`-method.

Can this be circumvented in order to mock a Page? Even current [tests](https://github.com/getgrav/grav/blob/1.7.0-beta.10/tests/unit/Grav/Common/Page/PagesTest.php#L77) do this. It would be beneficial to have easier access to spawning virtual Pages for both testing and dynamic processing.

As an example, the Error-plugin has to [spawn](https://github.com/getgrav/grav-plugin-error/blob/1.6.2/error.php#L60) from a [physical file](https://github.com/getgrav/grav-plugin-error/blob/1.6.2/pages/error.md) to dispatch. In similar cases, this is superfluous in both processing and handling, when minimal data is needed as a base for the Page-object.

**Comment by mahagr** [comment]
There is a plan to start deprecating methods from the old implementation and removing them from the interface -- for good. Right now the only reason why I opted from doing that is that it breaks all the automatic type hints in IDEs like PhpStorm and that is not very useful when I want to make the new class compatible with the old one.

This said Flex Pages do not implement `init()` method right now, they just throw exception `Not Implemented` when you try to call the method.

I am aware of the issue in the old Page class and there is a need to have some `SimplePage` class, which can be used for creating custom pages on the fly. Flex Page cannot be used for that either, as it requires the page being stored on somewhere -- though we could create custom memory storage for this.

We are also still missing a layer which allows you to combine pages from multiple sources to a single page structure; right now Pages has a hack to support both the old and new Flex pages. It does work, but it cannot really be optimized for speed/memory if you have a lot of pages.

What I suggest is to use some dummy `.md` file to initialize the page and then just override anything you want to. It is what admin and every plugin does...

**Comment by OleVik** [comment]
Yeah, that's what I'm doing right now. In-memory storage would be acceptable for Pages on the fly, and could also implement a `save()`-method for further storage if needed? It serves both the broad need for routing to non physical Pages, but also the narrow need for generating and storing a page without the need for a dummy.
